files/pull_repos.sh

```diff
@@ -0,0 +1,4 @@
+for dir in ~/work/*; 
```

Comment:
Add a shebang line for bash, just in case

Reply:
Shebang in this case is not needed since we are already in bash when calling this script.

Reply:
Can you explain how are we already in bash when calling this? 

While trivial change and this will work with every other shell, if we decide to extend the functionality of this in the future or the user chooses to do it themselves, its good to have shebang just in case.

Reply:
Shellbang is needed if you run the script directly by referencing the file, eg. `./shell-me-up.sh`. 
It really is a small change that has no downside

Reply:
Is it not being directly referenced inside the function?

Reply:
This is true, but if ever there is a reason to run `~/work/pull_repos.sh` and you really don't want to write `sh`, then it will not work without `#!/usr/bin/env bash`

Reply:
I remember reading once that if you explicitly **do not** wish the end user to run the script standalone, then you deliberately omit the shellbang. This is for shell scripts that are intended to be sourced by some other script. Being that pull_repos can be both, I would put shellbang.

Reply:
Okay, I agree, I did this with this project where hook-me-up.py has shebang at the top, but other scripts that are being imported don't have shebangs, is this the way you meant?

Reply:
@krukru @Dvorsky it can be ran already without shebang. Shebang is needed if we need to use different interpreter. By default it will run it as bash.

eg. if we want py script without typing "python ./something.py" then we would put shebang on the top.

second thing is, even if we put shebang, u can't run script manually since it is not executable. 
Ill add chmod +x command today so users can run it manually

Reply:
Wouldn't it use whatever your /bin/sh points at?

Reply:
If the file is chmoded to be executable, then it should also have a shellbang. 
Without the shellbang the OS will use whatever is in $SHELL (your default shell of choice). 

This script does work in bash and zsh, but surprise surprise it will not work in fish (without the shellbang)